File: src/main/java/tech/arhr/quingo/main_service/dto/QuizDto.java
package tech.arhr.quingo.main_service.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
public class QuizDto {
    private UUID id;

    @NotNull
    @Size(min = 2, max = 80)
    private String name;

    @Size(min = 10, max = 700)
    private String description;
}

File: src/main/kotlin/tech/arhr/quingo/dto/QuizDto.kt
package tech.arhr.quingo.dto

import java.util.UUID

data class QuizDto(
    val id: UUID,
    val ownerId: UUID,
    val visibility: Visibility,
    val status: QuizStatus,
    val draft: QuizContentDto? = null,
    val snapshot: QuizContentDto? = null,
)
